yellow-paper/docs/private-message-delivery/private-msg-delivery.md

```diff
@@ -19,20 +19,53 @@ If Alice executes a function that generates a note for Bob:
 
-The protocol will enable app developers to constrain the correctness of the following:
+The protocol will enable account contract developers to constrain the correctness of the following:
 
 1. The encryption of a user's note.
 2. The generation of the tag for that note.
 3. The publication of that note and tag to the correct data availability layer.
 
-Each app will define whether to constrain each such step. Encryption and tagging will be done through a set of [precompiled contracts](../addresses-and-keys/precompiles.md), each contract offering a different mechanism, and users will advertise their preferred mechanisms in a canonical [registry](./registry.md).
+Each account contract will define whether to constrain each such step. Encryption and tagging will be done through a set of [precompiled contracts](../addresses-and-keys/precompiles.md), each contract offering a different mechanism, and users will advertise their preferred mechanisms in a canonical [registry](../pre-compiled-contracts/registry.md).
 
 The advantages of this approach are:
 
-<!-- Perhaps we need a section justifying why a user should choose their note discovery preference vs an app, and all the painful discussions we had relating to that -->
-
-1. It enables a user to select their preferred [note discovery protocol](./note-discovery.md) and [encryption scheme](./encryption-and-decryption.md).
+1. It enables a user to select their preferred [note discovery protocol](./private-msg-delivery.md#note-discovery-protocol-selection) and [encryption scheme](./private-msg-delivery.md#encryption-and-decryption).
 2. It ensures that notes are correctly encrypted with a user's public encryption key.
-3. It ensures that notes are correctly tagged for a user's chosen [note discovery protocol](./note-discovery.md).
+3. It ensures that notes are correctly tagged for a user's chosen note discovery protocol.
 4. It provides scope for upgrading these functions or introducing new schemes as the field progresses.
 5. It protects applications from malicious unprovable functions.
+
+## Note Discovery Protocol Selection
+
+In order for a user to consume notes that belong to them, they need to identify, retrieve and decrypt them. A simple, privacy-preserving approach to this would be to download all of the notes and attempt decryption. However, the total number of encrypted notes published by the network will be substantial, making it infeasible for some users to do this. Those users will want to utilize a note discovery protocol to privately identify their notes.
+
+Selection of the encryption and tagging mechanisms to use for a particular note are the responsibilty of a user's account contract rather than the application generating the note. This is to ensure the note is produced in a way compatible with the user's chosen note discovery scheme. Leaving this decision to applications could result in user's having to utilise multiple note discovery schemes, a situation we want to avoid.
```

Comment:
```suggestion
Selection of the encryption and tagging mechanisms to use for a particular note are the responsibilty of a user's wallet rather than the application generating the note. This is to ensure the note is produced in a way compatible with the user's chosen note discovery scheme. Leaving this decision to applications could result in user's having to utilise multiple note discovery schemes, a situation we want to avoid.
```

The account contract does not signal the preferred method, it's the entry for the user in the registry.